enhancements/sha-bundle-digest.md

```diff
@@ -0,0 +1,80 @@
+
+---
+title: SHA-Bundle-Digest
+authors:
+- "@theishshah"
+reviewers:
+- @camilamacedo86 
+- @joelanford 
+- @jmrodri
+- @ryantking 
+- @tlwu2013 
+approvers:
+- @tlwu2013 
+- @jmrodri
+- @joelanford 
+creation-date: 2022-01-14
+last-updated: 2022-01-14
+status:implementable
+---
+
+# Title
+Add support for SHA-based Digest for Images
+
+## Summary
+
+Currently, operators generated by `operator-sdk` will store and retrieve container images based on the image name and the image tag in the bundle. This EP adds support to `operator-sdk` such that images can be stored and retrieved in the bundle with an image name and an image digest as an optional replacement for a tag. Image digests are SHA-256 sums that are hashed from the image layer therefore uniquely identifying one specific image.
+
+## Motivation
+
+Users in air-gapped (disconnected) environments are currently unable to get images based on tags, and can only access local registries based on the digest.
+
+This set of changes will allow SDK users in air-gapped (disconnected) environments to generate bundles with the image digest. This change is necessary due to non-deterministic behavior with images identified via tags, this issue specifically occurs in the OCP registry due to how the digest is generated with tag name as inputs.
+
+### Goals
+
+-   Give users the ability to the enable digest-based image resolution via a CLI flag when generating bundles.
+-   Allow users to specify additional images to resolve digests via an environment variable.
+-   Add the ability to resolve image digests from image tag dynamically using local docker registry.
+-   OPEN QUESTION: Store digest and tag mappings in file so that user-written code can resolve tags to digest.
```

Comment:
Move to the open questions section. I think this ties into our ongoing discussion of the cluster CSV annotations.